File: double-linked-list.py
class Node:
    def __init__ (self, data):
        self.data = data
        self.next = None
        self.prev = None

class DoubleLinkedLIst:
    def __init__ (self):
        self.head = None

    def append(self, data):
        if self.head is None:
            new_node = Node(data)
            new_node.prev = None
            self.head = new_node

        else:
            new_node = Node(data)
            cur = self.head
            while cur.next:
                cur = cur.next
            cur.next = new_node
            new_node.prev = cur
            new_node.next = None

    def prepend(self, data):
        if self.head is None:
            new_node = node(data)
            new_node.next = self.head
            self.head = new_node

        else:
            new_node = Node(data)
            self.head.prev = new_node
            new_node.next = self.head
            self.head = new_node
            new_node.prev = None

    def print_list(self):
        cur = self.head
        while cur:
            print(cur.data)
            cur = cur.next

d1 = int(input("Masukkan nilai : "))
d2 = int(input("Masukkan nilai : "))
d3 = int(input("Masukkan nilai : "))
d4 = int(input("Masukkan nilai : "))
d5 = int(input("Masukkan nilai : "))
d6 = int(input("Masukkan nilai : "))
e1 = int(input("Masukkan nilai : "))
e2 =  int(input("Masukkan nilai : "))



list = DoubleLinkedLIst()
list.append(d1)
list.append(d2)
list.append(d3)
list.append(d4)
list.append(d5)
list.append(d6)

list.prepend(e1)
list.prepend(e2)

list.print_list()

File: Java/DoubleLinkedList.java
class Node {
    String data;
    Node prev;
    Node next;
    void info() {
        System.out.println(" ");
        System.out.println("NILAI :"+this.data);
        System.out.println("POINTER "+this);
    }
}

public class DoubleLinkedList {
    static Node head = null, tail = null;
    static void push(Node node_baru) {
        if (head == null && tail == null) {
            head = node_baru;
            tail = node_baru;
        }else {
            node_baru.prev  = tail;
            tail.next = node_baru;
            tail = node_baru;
        }
    }
    public static void main(String[] args) {
        Node ana = new Node();
        ana.data= "Ana";
        Node Nana = new Node();
        Nana.data= "Nana";
        Node Fana = new Node();
        Fana.data= "Fana";
        Node Sana = new Node();
        Sana.data= "Sana";
        push(ana);
        push(Nana);
        push(Fana);
        push(Sana);
        head.info();
        tail.info();
        head.next.next.info();
    }
}